### PR TITLE
Spelling stdlib

### PR DESCRIPTION
**Pull request by jsoref**
This fixes some misspellings in Swift stdlib, it is split per https://github.com/apple/swift/pull/42421#issuecomment-1101092698

### CODE REVIEW

File: validation-test/stdlib/String.swift [hunk 1]
```diff
@@ -1859,7 +1859,7 @@ struct COWStringTest {
 }
 
 var testCases: [COWStringTest] {
-  return [ COWStringTest(test: "abcdefghijklmnopqrxtuvwxyz", name: "ASCII"),
+  return [ COWStringTest(test: "abcdefghijklmnopqrstuvwxyz", name: "ASCII"),
```
Comment (jsoref) [inline validation-test/stdlib/String.swift:1862]: ?!
```diff
-x
+s
```

File: test/stdlib/TestScanner.swift [hunk 2]
```diff
@@ -286,7 +286,7 @@ class TestScanner : TestScannerSuper {
     
     // Partial graphemes:
     withScanner(for: "e\u{0301}\u{031A}\u{032B}") {
-      // We do not assert here that the legacy methods don't work because they behave inconsistently wrt graphemes, and are able to actually discern that a combination code point plus a sequence of combining diacriticals is actually not OK to scan. Check just the newer behavior here.
+      // We do not assert here that the legacy methods don't work because they behave inconsistently wrt graphemes, and are able to actually discern that a combination code point plus a sequence of combining diacritics is actually not OK to scan. Check just the newer behavior here.
```
Comment (CodaFi) [inline test/stdlib/TestScanner.swift:289]: `combining diacriticals` is correct here.

File: test/stdlib/TestScanner.swift [hunk 3]
```diff
@@ -349,7 +349,7 @@ class TestScanner : TestScannerSuper {
     
     // Partial graphemes (diacritics):
     withScanner(for: "wow e\u{0301}\u{031A}\u{032B} NOT FOUND") {
-      // We do not assert here that the legacy methods don't work because they behave inconsistently wrt graphemes, and are able to actually discern that a combination code point plus a sequence of combining diacriticals is actually not OK to scan. Check just the newer behavior here.
+      // We do not assert here that the legacy methods don't work because they behave inconsistently wrt graphemes, and are able to actually discern that a combination code point plus a sequence of combining diacritics is actually not OK to scan. Check just the newer behavior here.
```
Comment (CodaFi) [inline test/stdlib/TestScanner.swift:352]: And here

**Review by Azoy — APPROVED** [review]
Thanks!

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Pull request merged**